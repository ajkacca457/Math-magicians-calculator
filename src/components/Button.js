import React from 'react';

const Button = props => {
  console.log(props);
  return (
    <>
      <button type="button">Hello</button>
    </>
  );
};

export default Button;
