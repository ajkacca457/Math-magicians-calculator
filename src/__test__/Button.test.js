import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from '../components/Button';

describe('Button Component', () => {
  it('renders correctly', () => {
    const btntree = TestRenderer
      .create(<Button />)
      .toJSON();
    expect(btntree).toMatchSnapshot();
  });

  const handleClick = () => { ''; };

  test('Checking if the button roles are button', () => {
    const { getByRole } = render(<Button name="=" wide={false} color={false} handleClick={handleClick} />);
    const eqlBtn = getByRole('button');
    expect(eqlBtn).toBeInTheDocument();
  });

  test('Checking if it renders the button texts correctly', () => {
    const { getByText } = render(<Button name="=" wide={false} color={false} handleClick={handleClick} />);
    const eqlBtn = getByText(/=/i);
    expect(eqlBtn).toBeInTheDocument();
  });
});
