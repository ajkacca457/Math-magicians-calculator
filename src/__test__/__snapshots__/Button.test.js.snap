// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  
</button>
`;
